spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/JdbcOutboundGateway.java

```diff
@@ -98,10 +97,54 @@ public JdbcOutboundGateway(JdbcOperations jdbcOperations, String updateQuery, St
 	 * This parameter is only applicable if a selectQuery was provided. Null values
 	 * are not permitted.
 	 * @param maxRowsPerPoll the number of rows to select. Must not be null.
+	 * @deprecated since 5.1 in favor of {@link #setMaxRows(Integer)}
 	 */
+	@Deprecated
 	public void setMaxRowsPerPoll(Integer maxRowsPerPoll) {
-		Assert.notNull(maxRowsPerPoll, "MaxRowsPerPoll must not be null.");
-		this.maxRowsPerPoll = maxRowsPerPoll;
+		setMaxRows(maxRowsPerPoll);
+	}
+
+	/**
+	 * The maximum number of rows to query.
+	 * The value is ultimately set on the underlying {@link JdbcPollingChannelAdapter}.
+	 * If not specified this value will default to {@code 1}.
+	 * This parameter is only applicable if a selectQuery was provided. Null values
+	 * are not permitted.
+	 * @param maxRows the number of rows to select. Must not be null.
+	 * @since 5.1
+	 * @see JdbcPollingChannelAdapter#setMaxRows(int)
+	 */
+	public void setMaxRows(Integer maxRows) {
```

Comment:
Why not just `int` here? Null check not needed then (but restore the `Assert` in the deprecated method).

Reply:
Well, there is a logic in the `onInit()`:
```
if (this.maxRows != null) {
		Assert.notNull(this.poller, "If you want to set 'maxRows', then you must provide a 'selectQuery'.");
	this.poller.setMaxRows(this.maxRows);
}
```
Should we change this to something like `> 0` then?
Also pay attention to the logic in the ctor:
```
if (StringUtils.hasText(selectQuery)) {
	this.poller = new JdbcPollingChannelAdapter(jdbcOperations, selectQuery);
	this.poller.setMaxRows(1);
}
```
The point is that with the regular poller we have an unlimited request, but with the gateway we are limiting to only the `1` by default. Any custom value is managed by the `Integer` state.

Reply:
Agreed; I missed that; the current state is fine.